watcher/watcher.go

```diff
@@ -803,14 +807,12 @@ func (w *Watcher) AutoScale() error {
 			w.Run = false
 			return err
 		}
-		logger.WithError(err).Error("failed to get scheduler scaling info")
-		return err
+		return fmt.Errorf("failed to get scheduler scaling info: %s", err)
```

Comment:
Maybe you wanted to use %w instead of %s here. Or maybe add .Error() to InvalidPodWaitingStateErr and replace fmt.Errorf by errors.New + fmt.Sprintf.